### PR TITLE
Create scheduled report with zero objects selectable

### LINKED ISSUE #3886: Live set without objects cannot choose report types
**Describe the bug**
Currently it is not possible to select report types (step 2 of the report-flow) if the selected filters do not result in any objects.

**To Reproduce**

1. Create a new organization (without objects)
2. Go to reports and click "Generate Report"
3. Click on "Continue with live set"
4. Report types are 0

**Expected behavior**
When a new organization is created, it should be possible to create a scheduled report, even when there are no objects yet. This is because these objects might be generated in the future.

Note: this means that we should also catch the case where a scheduled report is generated while there are 0 objects. This should return in an empty report and should be clear to the user why this report is empty.

**Screenshots**
Image

**OpenKAT version**
main

### PR DESCRIPTION
**Pull request by madelondohmen**
### Changes
This PR makes the following possible:
- A user can always see and select report types, even when there are no objects that meet the live filter
   - The report types that are visible are based on the ooi_types that the user selected in the filter
- A user can now create a scheduled report based on a live set without objects
- The empty reports with 0 input objects will be generated and showed on the Scheduled Reports Page, but not on the History Page (this is in line with a future design).

### Issue link
Closes #3886

### Demo

https://github.com/user-attachments/assets/ad78b7a6-d0e1-4601-87c0-4079e2f0664b

afbeelding

### QA notes
- Create a normal Report and Aggregate Report
- Change the filter so that no objects match the filter
- You should still be able to continue
- No empty reports should be visible on the Report History page
- The empty reports should be visible (but not clickable) on the Scheduled Reports page

---

### Code Checklist

- [ ] All the commits in this PR are properly PGP-signed and verified.
- [ ] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] I have checked the documentation and made changes where necessary.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/reports/views/base.py [hunk 1]
```diff
@@ -494,18 +499,18 @@ def create_report_names(oois: list[OOI], report_types: list[type[BaseReport]]) -
...
         oois_count = len(oois)
         report_types_count = len(report_types)
-        ooi = oois[0].human_readable
+        ooi = oois[0].human_readable if oois else None
         report_type = report_types[0].name
 
         # Create name for parent report
         if not (report_types_count == 1 and oois_count == 1):
-            if report_types_count > 1 and oois_count > 1:
+            if report_types_count > 1 and (oois_count == 0 or oois_count > 1):
```
Comment (ammar92) [inline rocky/reports/views/base.py:507]:
```suggestion
            if report_types_count > 1 and oois_count != 1):
```

File: rocky/reports/views/base.py [hunk 2]
```diff
@@ -494,18 +499,18 @@ def create_report_names(oois: list[OOI], report_types: list[type[BaseReport]]) -
...
                 name = _("Concatenated Report for {oois_count} objects").format(
                     report_type=report_type, oois_count=oois_count
                 )
             elif report_types_count > 1 and oois_count == 1:
                 name = _("Concatenated Report for {ooi}").format(ooi=ooi)
-            elif report_types_count == 1 and oois_count > 1:
+            elif report_types_count == 1 and (oois_count == 0 or oois_count > 1):
```
Comment (ammar92) [inline rocky/reports/views/base.py:513]:
```suggestion
            elif report_types_count == 1 and oois_count != 1):
```

File: rocky/reports/report_types/helpers.py [hunk 3]
```diff
@@ -59,10 +59,23 @@ def get_report_types_for_oois(oois: list[str]) -> set[type[BaseReport]]:
...
+def get_report_types_for_ooi_type(ooi_type: str) -> list[type[BaseReport]]:
+    """
+    Get all report types that can be generated for a given OOI Type
+    """
+    return [report for report in REPORTS if ooi_type in report.input_ooi_types]
+
+
+def get_report_types_for_ooi_types(ooi_types: list[str]) -> set[type[BaseReport]]:
+    """
+    Get all report types that can be generated for a given list of OOIs
```
Comment (ammar92) [inline rocky/reports/report_types/helpers.py:74]:
```suggestion
    Get all report types that can be generated for a given list of OOI types
```

File: rocky/reports/report_types/helpers.py [hunk 4]
```diff
@@ -59,10 +59,23 @@ def get_report_types_for_oois(oois: list[str]) -> set[type[BaseReport]]:
     """
     Get all report types that can be generated for a given list of OOIs
     """
-
     return {report for ooi_pk in oois for report in get_report_types_for_ooi(ooi_pk)}
 
 
+def get_report_types_for_ooi_type(ooi_type: str) -> list[type[BaseReport]]:
+    """
+    Get all report types that can be generated for a given OOI Type
```
Comment (ammar92) [inline rocky/reports/report_types/helpers.py:67]:
```suggestion
    Get all report types that can be generated for a given OOI type
```

File: rocky/reports/report_types/helpers.py [hunk 5]
```diff
@@ -59,10 +59,23 @@ def get_report_types_for_oois(oois: list[str]) -> set[type[BaseReport]]:
     """
     Get all report types that can be generated for a given list of OOIs
     """
-
     return {report for ooi_pk in oois for report in get_report_types_for_ooi(ooi_pk)}
 
 
+def get_report_types_for_ooi_type(ooi_type: str) -> list[type[BaseReport]]:
```
Comment (ammar92) [inline rocky/reports/report_types/helpers.py:65]: Any reason why this function returns a list, and `get_report_types_for_ooi_types` a set (or vice versa)?

**Review by ammar92 — COMMENTED** [review]
Tiny remarks, looks good otherwise

**Review by underdarknl — APPROVED** [review]

**Pull request merged**